backend/tardigrade/fs.go

```diff
@@ -407,16 +410,6 @@ func (f *Fs) listObjects(ctx context.Context, relative, bucketName, bucketPath s
 		return nil, err
 	}
 
-	// NOTE: This is necessary because libuplink does not support listings
-	// without a trailing slash `/`. This means you can't call list on an
-	// actual object. However, rclone expects you to be able to do this. To
-	// bridge the gap we do a stat on the base of the listing and add it to
-	// our results if necessary.
-	object, err := f.project.StatObject(ctx, bucketName, bucketPath)
-	if err == nil {
-		entries = append(entries, f.newDirEntry(relative, opts.Prefix, object))
-	}
-
```

Comment:
How does removing this relates to the object as root thing?

Reply:
https://github.com/rclone/rclone/pull/3699/files#r418069892

Per the maintainer, this shouldn't ever happen except when the filesystem is mounted to an object. And that is supposed to be handled in the NewFs setup, but I was doing it wrong before. By handling that correctly in NewFs we no longer have a situation where it expects the root itself to be in the listing.